### PR TITLE
TCTI refactor part 1

### PR DESCRIPTION
**Pull request by flihp**
This is the "low hanging fruit" in the TCTI modules. Most of this is re-organizing some of the build bits. The rest implements a common write function shared between the device and socket TCTIs. This is easier than messing with the read path since it's not an async call. The build structure introduced here will make writing and testing the async read stuff easier though.

### CODE REVIEW

File: tcti/tcti.h [hunk 1]
```diff
@@ -52,6 +55,13 @@
 
 #define TCTI_CONTEXT ((TSS2_TCTI_CONTEXT_COMMON_CURRENT *)(SYS_CONTEXT->tctiContext))
 
+#define TEMP_RETRY(exp) \
+({  int __ret; \
+    do { \
+        __ret = exp; \
+    } while (__ret == -1 && __ret == EINTR); \
```
Comment (tstruk) [inline tcti/tcti.h]: Should be while (__ret == -1 || __ret == EINTR);

File: tcti/tcti.c [hunk 2]
```diff
@@ -96,3 +100,31 @@ TSS2_RC tcti_receive_checks (
...
+                  fd);
+        written = TEMP_RETRY (write (fd,
+                                     (const char*)&buf [written_total],
+                                     size - written_total));
+        if (written >= 0) {
+            LOG_DEBUG ("wrote %zd bytes to fd %d", written, fd);
+            written_total += (size_t)written;
+        } else {
+            LOG_ERROR ("failed to write to fd %d: %s", fd, strerror (errno));
+            return written_total;
```
Comment (tstruk) [inline tcti/tcti.c]: return written_total ? written_total : -1;
According to man 3 write 0 is not a valid return value from write(). It should be either number of bytes written (if any) or -1.

File: tcti/platformcommand.c [hunk 3]
```diff
@@ -50,16 +43,17 @@ TSS2_RC PlatformCommand(
 {
     TSS2_TCTI_CONTEXT_INTEL *tcti_intel = tcti_context_intel_cast (tctiContext);
     int iResult = 0;            // used to return function results
-    char sendbuf[] = { 0x0,0x0,0x0,0x0 };
+    uint8_t sendbuf[] = { 0x0,0x0,0x0,0x0 };
```
Comment (tstruk) [inline tcti/platformcommand.c]: add space after commas like it is in the next line

File: include/tcti/common.h [hunk 4]
```diff
@@ -1,43 +1,36 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
 #ifndef TCTI_COMMON_H
 #define TCTI_COMMON_H
 
-#include "sapi/tpm20.h"
-
 #if defined (__GNUC__)
```
Comment (tstruk) [inline include/tcti/common.h]: Do we need this file at all now?
Looks like the COMPILER_ATTR is also (re)defined in log/log.h.
Let's put it to include/sapi/tpm20.h and get rid of the include/tcti/common.h altogether.

File: tcti/tcti.c (hunk 2 above)
Comment (flihp) [reply tcti/tcti.c]: I think you want `man 2 write`. Linux allows for a 0 return from `write` when nothing is written.

File: include/tcti/common.h (hunk 4 above)
Comment (flihp) [reply include/tcti/common.h]: This is one of the eventual goals but I'm deferring this till a later cleanup patch. Hopefully things will be a bit more clear after some more internal cleanup.

**Comment by flihp** [comment]
Thanks for the quick review @tstruk. Fixups pushed addressing everything except your comment w/r to `write` and the possibility of a 0 return value. Let me know how you want to proceed on that one.

**Pull request merged**